acceptance/lib/puppet/acceptance/rpm_util.rb

```diff
@@ -15,10 +16,11 @@ def rpm_provider(agent)
 
       def setup(agent)
         cmd = rpm_provider(agent)
-        required_packages = ['createrepo', 'rpm-build']
+        required_packages = ['createrepo', 'curl', 'rpm-build']
```

Comment:
@Iristyle similarly to the code you changed here, I think the code you referenced in beaker is doing an existence check and installing if it's not there, not a version check.

I think the utilities provided here are [used seldom enough](https://github.com/search?utf8=✓&q=org%3Apuppetlabs+send_rpm+extension%3Arb&type=Code&ref=advsearch&l=&l=) that they alone shouldn't create enough need for beaker to always be checking against the right versions, especially since for most things, package existence is all we really need.

Beaker does provide the ability to upgrade packages with:
- [Beaker::Host's `upgrade_package`](https://github.com/puppetlabs/beaker/blob/master/lib/beaker/host/unix/pkg.rb#L195)
- [DSL convenience method: `upgrade_package`](https://github.com/puppetlabs/beaker/blob/master/lib/beaker/dsl/helpers/host_helpers.rb#L411)

So in general I would recommend using those for upgrades, but this file is already low-level enough that you might as well just skip to making the package install call yourself, as you are.

Another option would be to specify in the beaker code you referenced the specific versioned package of curl for fedora, like freebsd does [here](https://github.com/puppetlabs/beaker/blob/master/lib/beaker/host_prebuilt_steps.rb#L16) (although they allow a backup of the unversioned package).